### DISCUSSION TITLE
using vscode with USB connection?

### DISCUSSION THREAD

**Discussion by deopark** [pybricks/support#1976] [Q&A] [answered]
I have finished setting up hub(spike Prime) with vscode.
It works well with BLE.

But we sometimes want to download and run with USB connection.
Do you know how to download and run with USB connection in vscode?

Below is the launch.json I'm using and the error message.
launch.json
{
            "name": "Run USB",
            "type": "debugpy",
            "request": "launch",
            "module": "pybricksdev",
            "console": "integratedTerminal",
            "args": ["run", "usb", "${file}"],
}

result 
raise OSError("Could not find hub.")
OSError: Could not find find hub

thank you.

**Comment by dlech** [comment, marked as answer]
We don't have USB support yet. We are working on it, but it hasn't been the top priority. Keep an eye on https://github.com/pybricks/pybricks-micropython/pull/208
